src/Views/Bookmark.elm

```diff
@@ -0,0 +1,268 @@
+module Views.Bookmark exposing (ActiveTab(..), Scope(..), view)
+
+import Data.Bookmark as Bookmark exposing (Bookmark)
+import Data.Impact as Impact
+import Data.Session exposing (Session)
+import Data.Unit as Unit
+import Html exposing (..)
+import Html.Attributes exposing (..)
+import Html.Events exposing (..)
+import Page.Textile.Simulator.ViewMode as ViewMode exposing (ViewMode)
+import Route
+import Time
+import Views.Icon as Icon
+
+
+type alias ManagerConfig msg =
+    { session : Session
+    , activeTab : ActiveTab
+    , bookmarkName : String
+    , impact : Impact.Definition
+    , funit : Unit.Functional
+    , viewMode : ViewMode
+    , scope : Scope
+
+    -- Messages
+    , copyToClipBoard : String -> msg
+    , compare : msg
+    , delete : Bookmark -> msg
+    , save : msg
+    , update : String -> msg
+    , switchTab : ActiveTab -> msg
+    }
+
+
+type Scope
+    = Food
+    | Textile
+
+
+type ActiveTab
+    = SaveTab
+    | ShareTab
+
+
+view : ManagerConfig msg -> Html msg
+view ({ activeTab, switchTab } as config) =
+    div [ class "card shadow-sm" ]
+        [ div [ class "card-header" ]
+            [ [ ( SaveTab, "Sauvegarder" ), ( ShareTab, "Partager" ) ]
+                |> List.map
+                    (\( msg, label ) ->
+                        li [ class "nav-item" ]
+                            [ button
+                                [ class "btn btn-text nav-link rounded-0 rounded-top no-outline"
+                                , classList [ ( "active", activeTab == msg ) ]
+                                , onClick <| switchTab msg
+                                ]
+                                [ text label ]
+                            ]
+                    )
+                |> ul [ class "nav nav-tabs justify-content-end card-header-tabs" ]
+            ]
+        , case activeTab of
+            ShareTab ->
+                shareLinkView config
+
+            SaveTab ->
+                managerView config
+        ]
+
+
+shareLinkView : ManagerConfig msg -> Html msg
+shareLinkView { session, impact, funit, copyToClipBoard, scope } =
+    let
+        shareableLink =
+            case scope of
+                Food ->
+                    Just session.queries.food
+                        |> Route.FoodBuilder impact.trigram
+                        |> Route.toString
+                        |> (++) session.clientUrl
```

Comment:
Should we have a `Session.Scope` and a `Session.queryToUrl`?

Reply:
But what if we ever want to render textile and food bookmarks on a same page? 🤔